presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2142,9 +2165,9 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
             return Optional.empty();
         }
 
-        return Optional.of(new HivePartitioningHandle(
+        return Optional.of(createPartitioningHandleWithHiveType(
```

Comment:
Let's keep it compatible with native partitioning handle as well. 
Please check if the partitioning handle type is the same and if the column types are the same as well.  Also it looks like it makes sense to keep the constructor public, so it is easier to change the `maxCompatibleBucketCount` without performing extra logic.

Reply:
Good point, I forget to check column types consistency. This method is marked as deprecated but I think the code is still in use, right?